src/core/fetchers/segment/prioritizer.ts

```diff
@@ -224,55 +227,114 @@ export default class ObservablePrioritizer<T> {
-        // This task doesn't have priority yet. Start it on trigger
         this._waitingQueue.push(newTask);
-        return trigger.pipe(switchMap(onTrigger));
+      } else {
+        newTask.trigger(true);
+        if (this._isRunningHighPriorityTasks()) {
+          // Note: we want to begin interrupting low-priority tasks just
+          // after starting the current one because the interrupting
+          // logic can call external code.
+          // This would mean re-entrancy, itself meaning that some weird
+          // half-state could be reached unless we're very careful.
+          // To be sure no harm is done, we put that code at the last
+          // possible position (the previous Observable sould be
+          // performing all its initialization synchronously).
+          this._interruptCancellableTasks();
+        }
       }
 
-      // Start it right away
-      const startTask$ = trigger.pipe(startWith(true), switchMap(onTrigger));
-      return !this._isHighPriority(newTask) ?
-        startTask$ :
-        observableMerge(startTask$,
-
-                        // Note: we want to begin interrupting low-priority tasks just
-                        // after starting the current one because the interrupting
-                        // logic can call external code.
-                        // This would mean re-entrancy, itself meaning that some weird
-                        // half-state could be reached unless we're very careful.
-                        // To be sure no harm is done, we put that code at the last
-                        // possible position (the previous Observable sould be
-                        // performing all its initialization synchronously).
-                        observableDefer(() => {
-                          this._interruptCancellableTasks();
-                          return EMPTY;
-                        }));
+      /** Callback called when this Observable is unsubscribed to. */
+      return () => {
+        isStillSubscribed = false;
+        if (newTask.subscription !== null) {
+          newTask.subscription.unsubscribe();
+          newTask.subscription = null;
+        }
+
+        if (newTask.finished) { // Task already finished, we're good
+          return;
+        }
+
+        // remove it from waiting queue if in it
+        const waitingQueueIndex = arrayFindIndex(this._waitingQueue,
+                                                 (elt) => elt.observable === pObs$);
+
+        if (waitingQueueIndex >= 0) { // If it was still waiting for its turn
+          this._waitingQueue.splice(waitingQueueIndex, 1);
+        } else {
+          // remove it from pending queue if in it
+          const pendingTasksIndex = arrayFindIndex(this._pendingTasks,
+                                                   (elt) => elt.observable === pObs$);
+          if (pendingTasksIndex < 0) {
+            log.warn("FP: unsubscribing non-existent task");
```

Comment:
Is it really necessary to warn the user about the non-existent task ? I mean, user may not understand the subject of this log. Maybe it could be better as a debug log, no ?

Reply:
To me, logs are both for application developers and us.
This case should never ever be encountered even if it is not an error per se and could mean there's an underlying problem.

That's why I prefer to add visibility to it so that an application dev can see it.
With chance it could help that person reporting a bug to us.

Reply:
For me logs are both for application devs and us.

This case should never ever be encountered even if it is not an error if it is. Seeing it could mean that an underlying problem exist.

That's why I prefer to put more visibility to it, even set it as a "warning" because that's what it is.

It would be nice to have this in a bug report if it has been encountered, for example.

But yes, it's true that this sentence most likely won't make any sense to someone not familiar to the RxPlayer's code.